### ISSUE TITLE
Rockstar does not work with Python 3

### ISSUE THREAD

**Issue by brittonsmith** [yt-project/yt_astro_analysis#18]
The `RockstarHaloFinder` does not seem to work in Python 3.  Below are a test script and the error output produced.

## Script
```
import yt
from yt_astro_analysis.halo_finding.rockstar.api import \
    RockstarHaloFinder
from yt.data_objects.particle_filters import \
    particle_filter
yt.enable_parallelism()

@particle_filter("dark_matter", requires=["creation_time"])
def _dm_filter(pfilter, data):
    return data["creation_time"] <= 0.0

def setup_ds(ds):
    ds.add_particle_filter("dark_matter")

es = yt.simulation("enzo_tiny_cosmology/32Mpc_32.enzo", "Enzo")
es.get_time_series(setup_function=setup_ds,
                   redshifts=[1., 0.])

rh = RockstarHaloFinder(es, num_readers=1, num_writers=1,
                        particle_type="dark_matter")
rh.run()
```
Run with: `mpirun -np 3 python <script>`

## Output
```
[Warning] Connection attempt 1 to bunkbed:47857 failed: : Connection refused
[Error] Couldn't open strict:!  (Err: Name or service not known)
-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code.. Per user-direction, the job has been aborted.
-------------------------------------------------------
[Warning] Connection attempt 1 to bunkbed:47857 failed: : Connection refused
```

**Comment by ngoldbaum** [comment]
Kacper points out on the yt-dev mailing list that this might have something to do with objects getting garbage-collected after they are passed to rockstar and offers this patch:

https://paste.pound-python.org/show/QdIThLDuq3YcqRnzwrED/

Apparently with this the clients are still unable to contact the server, but with this patch the clients retain the correct address and port number for the server.

**Comment by brittonsmith** [comment]
With Kacper's solution, I was able to get clients connecting to the server, but no halos are found when they are in Python 2. I will start a branch in my fork to keep track of the progress on this.

**Comment by brittonsmith** [comment]
Fixed in [yt PR #1740](https://github.com/yt-project/yt/pull/1740).
